src/StreamFactoryInterface.php

```diff
@@ -9,15 +9,11 @@
     /**
      * Create a new stream from a resource or a string.
      *
-     * If the argument is a resource, it MUST be readable and SHOULD be seekable. It MAY be writable.
-     * If the argument is a string, a temporary resource will be created that is writable and seekable.
-     * 
-     * If the arugment is a string it will be interpided as the content of the stream. File names or 
-     * file paths are not supported. 
+     * String content will be used as the body of the stream. Files paths are not supported.
```

Comment:
> Would it make sense to modify @mindplay-dk's suggestion by adding an interface?

It would make sense only in terms of convenience.

Per my example above, that interface does not provide any additional interoperability - the other interfaces, upon which this one would strictly depend, are providing the actual contract; this interface defines a facility for a layer of abstraction on top of those contracts.

Strictly speaking, it complicates interoperability, by requiring every implementation to re-implement the exact same layer of abstraction I posted in source-code above. This can just as well be a package you install - which works with any implementation, rather than requiring each implementation to recreate that layer.

Whether you even *want* such a layer, is a matter of preference. I'd be fine without it, and clearly you would not, but you could then install a package that provides that facility, still be fully-interoperable, without complicating the standard, and without duplicating that facility in each implementation.